lua/telescope/actions/init.lua

```diff
@@ -143,6 +144,50 @@ function actions.toggle_selection(prompt_bufnr)
   current_picker:toggle_selection(current_picker:get_selection_row())
 end
 
+--- Multi select all entries.
+--- - Note: selected entries may include results not visible in the results popup.
+---@param prompt_bufnr number: The prompt bufnr
+function actions.select_all(prompt_bufnr)
+  local current_picker = action_state.get_current_picker(prompt_bufnr)
+  -- indices are 1-indexed, rows are 0-indexed
+  local select_all = function(entry, _, row)
+    if not current_picker._multi:is_selected(entry) then
+      current_picker._multi:add(entry)
+      if current_picker:can_select_row(row) then
+        current_picker.highlighter:hi_multiselect(row, current_picker._multi:is_selected(entry))
+      end
+    end
+  end
+  action_utils.map_entries(prompt_bufnr, select_all)
+end
+
+--- Drop all entries from the current multi selection.
+---@param prompt_bufnr number: The prompt bufnr
+function actions.drop_all(prompt_bufnr)
+  local current_picker = action_state.get_current_picker(prompt_bufnr)
+  local drop_all = function(entry, _, row)
+    current_picker._multi:drop(entry)
+    if current_picker:can_select_row(row) then
+      current_picker.highlighter:hi_multiselect(row, current_picker._multi:is_selected(entry))
+    end
+  end
+  action_utils.map_entries(prompt_bufnr, drop_all)
+end
+
+--- Toggle multi selection for all entries.
+--- - Note: toggled entries may include results not visible in the results popup.
+---@param prompt_bufnr number: The prompt bufnr
+function actions.toggle_all(prompt_bufnr)
+  local current_picker = action_state.get_current_picker(prompt_bufnr)
+  local toggle_all = function(entry, _, row)
+    current_picker._multi:toggle(entry)
+    if current_picker:can_select_row(row) then
+      current_picker.highlighter:hi_multiselect(row, current_picker._multi:is_selected(entry))
+    end
+  end
+  action_utils.map_entries(prompt_bufnr, toggle_all)
```

Comment:
can you just do 
```lua
actions_utils.map_entries(prompt_bufnr, function(entry, _, row)

end)
```

applies to all your actions. no reason to have it as local function

Reply:
Genuinely curious, is there a difference apart from cosmetics and the function then not only being in scope of the `map_entries` call?

Reply:
nope. in my opinion its just more readable. Its like a callback/lambda in any other language